File: src/main/java/com/gamestore/entity/VideoJuego.java
package com.gamestore.entity;

import jakarta.persistence.*;
import lombok.*;
import java.util.List;

@Entity
@Table(name = "videojuegos")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class VideoJuego {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idVideojuego;

    private String titulo;
    private String descripcion;
    private String plataforma;
    private Double precio;
    private Integer stock;

    @Column(name = "imagen_url")
    private String imagenUrl;

    @OneToMany(mappedBy = "videojuego", cascade = CascadeType.ALL)
    private List<Carrito_Items> carritoItems;

    @OneToMany(mappedBy = "videojuego", cascade = CascadeType.ALL)
    private List<Pedido_Items> pedidoItems;
}

File: web/app/Models/Videojuego.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Videojuego extends Model
{
    protected $table = 'videojuegos';
    protected $primaryKey = 'id_videojuego';
    public $timestamps = false;

    protected $fillable = ['titulo', 'descripcion', 'plataforma', 'precio', 'stock', 'imagen_url'];
}
